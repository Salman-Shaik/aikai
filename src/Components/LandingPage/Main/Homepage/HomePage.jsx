import _ from "lodash";
import React from "react";
import "../../../../css/HomePage.css";
import { IntroductionPage } from "./IntroductionPage";

const InfoSection = ({ description, onClick }) => {
  return (
    <section className="info_section">
      <p className="page_info">{description}</p>
      <button type="button" className="visit" onClick={onClick}>
        Visit â†’
      </button>
    </section>
  );
};

const Page = ({ keyword, image, alt, description, onClick }) => {
  return (
    <section id={keyword} className="page">
      <section className={keyword}>
        <img src={image} alt={alt} className={`${keyword}_image`} />
        <InfoSection description={description} onClick={onClick} />
      </section>
    </section>
  );
};

const NowPlayingPage = ({ updateLocation }) => (
  <Page
    keyword="now-playing"
    image="nowPlaying.jpg"
    alt="Now Playing"
    description="Wanna know what's playing in Theatres or which TV show is airing today."
    onClick={() => updateLocation("/now_playing")}
  />
);

const ShowPage = ({ image1, image2, description, updateLocation, type }) => {
  return (
    <section className="show-page">
      <section className="examples">
        <img
          src={image1}
          alt={`Top Rated ${_.capitalize(type)}`}
          className="show_example"
        />
        <img
          src={image2}
          alt={`Random ${_.capitalize(type)}`}
          className="show_example"
        />
      </section>
      <InfoSection description={description} onClick={() => updateLocation()} />
    </section>
  );
};

const MoviesPage = ({ updateLocation }) => (
  <ShowPage
    image1="movies-top.jpg"
    image2="movies-random.jpg"
    description="Puzzled which Movie to watch? We are here to help you."
    updateLocation={() => updateLocation("/movies")}
    type="movie"
  />
);

const TVPage = ({ updateLocation }) => (
  <ShowPage
    image1="tv-top.jpg"
    image2="tv-random.jpg"
    description="Want to watch a TV / Web Series. We got your back with suggestions."
    updateLocation={() => updateLocation("/tv_shows")}
    type="tv"
  />
);

const EditorsChoicePage = ({ updateLocation }) => (
  <Page
    keyword="editors_choice"
    image="editorsChoice.jpg"
    alt="Editor's Choice"
    description="Recommendations from the Editor, he says these are best he has ever seen. Do you believe him?"
    onClick={() => updateLocation("/editors_choice")}
  />
);

const ActionsPage = ({ updateLocation, action, description }) => {
  return (
    <section id={action} className="page">
      <section className={action}>
        <img src={`${action}.jpg`} alt={action} className="action_image" />
        <InfoSection
          description={description}
          onClick={() => updateLocation()}
        />
      </section>
    </section>
  );
};

const FavoritesPage = ({ updateLocation }) => {
  return (
    <ActionsPage
      updateLocation={() => updateLocation("/favorite_shows")}
      description="You can add a movie or a tv shows to your favorites list but you need to login first."
      action="favorites"
    />
  );
};

const WatchListPage = ({ updateLocation }) => {
  return (
    <ActionsPage
      updateLocation={() => updateLocation("/watch_list")}
      description="You can keep a track of all the movies or tv shows which you want to watch in future but you need to login first."
      action="watchlist"
    />
  );
};

const WaveComponent = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1440 320">
      <path
        fill="#EB2F53"
        fillOpacity="1"
        d="M0,64L48,58.7C96,53,192,43,288,69.3C384,96,480,160,576,160C672,160,768,96,864,106.7C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      />
    </svg>
  );
};

const Footer = () => {
  return (
    <section className="feature_page">
      <section className="features">
        <span className="features_header">Features coming soon</span>
        <ul className="feature_list">
          <li className="feature">Subscriptions</li>
          <li className="feature">New Features are yet to add</li>
        </ul>
      </section>
      <section className="feedback">
        <span className="feedback_header">
          Feedback makes A.I.K.A.I perfect
        </span>
        <button
          type="button"
          className="feedback_button"
          onClick={() =>
            window.open("https://forms.gle/MAiEH8fJDmZc6oda6", "_blank")
          }
        >
          Share feedback
        </button>
      </section>
      <span className="signature">
        Made by{" "}
        <a
          href="https://github.com/Salman-Shaik"
          target="_blank"
          rel="noopener noreferrer"
          className="profile_link"
        >
          Salman Shaik
        </a>
      </span>
      <WaveComponent />
    </section>
  );
};

export const HomePage = ({ updateLocation }) => {
  return (
    <section className="homepage">
      <IntroductionPage />
      <NowPlayingPage updateLocation={updateLocation} />
      <MoviesPage updateLocation={updateLocation} />
      <TVPage updateLocation={updateLocation} />
      <EditorsChoicePage updateLocation={updateLocation} />
      <FavoritesPage updateLocation={updateLocation} />
      <WatchListPage updateLocation={updateLocation} />
      <Footer />
    </section>
  );
};
