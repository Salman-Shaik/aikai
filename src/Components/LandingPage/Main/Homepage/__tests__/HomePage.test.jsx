import {render} from "@testing-library/react";
import React from "react";
import {HomePage} from "../HomePage";

describe("Homepage", () => {
  it("Snapshot Test", () => {
    const {container} = render(<HomePage/>);
    expect(container).toMatchSnapshot();
  });
})