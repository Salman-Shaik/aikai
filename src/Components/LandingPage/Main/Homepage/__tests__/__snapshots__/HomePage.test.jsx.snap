// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Snapshot Test 1`] = `
<div>
  <section
    class="homepage"
  >
    <section
      class="introduction_page"
    >
      <section
        class="introduction"
      >
        <span
          class="heading"
        >
          Random Movie / TV Show Generator
        </span>
        <p
          class="info"
        >
          Confused what to watch today, want to know the details of a movie/tv show, want to get recommendations for a movie/tv show, want to know what's playing in theatres or what's airing on TV. You can get all of them here @ A.I.K.A.I .
        </p>
      </section>
    </section>
    <section
      class="page"
      id="now-playing"
    >
      <section
        class="now-playing"
      >
        <img
          alt="Now Playing"
          class="now-playing_image"
          src="nowPlaying.jpg"
        />
        <section
          class="info_section"
        >
          <p
            class="page_info"
          >
            Wanna know what's playing in Theatres or which TV show is airing today.
          </p>
          <button
            class="visit"
            type="button"
          >
            Visit →
          </button>
        </section>
      </section>
    </section>
    <section
      class="show-page"
    >
      <section
        class="examples"
      >
        <img
          alt="Top Rated Movie"
          class="show_example"
          src="movies-top.jpg"
        />
        <img
          alt="Random Movie"
          class="show_example"
          src="movies-random.jpg"
        />
      </section>
      <section
        class="info_section"
      >
        <p
          class="page_info"
        >
          Puzzled which Movie to watch? We are here to help you.
        </p>
        <button
          class="visit"
          type="button"
        >
          Visit →
        </button>
      </section>
    </section>
    <section
      class="show-page"
    >
      <section
        class="examples"
      >
        <img
          alt="Top Rated Tv"
          class="show_example"
          src="tv-top.jpg"
        />
        <img
          alt="Random Tv"
          class="show_example"
          src="tv-random.jpg"
        />
      </section>
      <section
        class="info_section"
      >
        <p
          class="page_info"
        >
          Want to watch a TV / Web Series. We got your back with suggestions.
        </p>
        <button
          class="visit"
          type="button"
        >
          Visit →
        </button>
      </section>
    </section>
    <section
      class="page"
      id="editors_choice"
    >
      <section
        class="editors_choice"
      >
        <img
          alt="Editor's Choice"
          class="editors_choice_image"
          src="editorsChoice.jpg"
        />
        <section
          class="info_section"
        >
          <p
            class="page_info"
          >
            Recommendations from the Editor, he says these are best he has ever seen. Do you believe him?
          </p>
          <button
            class="visit"
            type="button"
          >
            Visit →
          </button>
        </section>
      </section>
    </section>
    <section
      class="page"
      id="favorites"
    >
      <section
        class="favorites"
      >
        <img
          alt="favorites"
          class="action_image"
          src="favorites.jpg"
        />
        <section
          class="info_section"
        >
          <p
            class="page_info"
          >
            You can add a movie or a tv shows to your favorites list but you need to login first.
          </p>
          <button
            class="visit"
            type="button"
          >
            Visit →
          </button>
        </section>
      </section>
    </section>
    <section
      class="page"
      id="watchlist"
    >
      <section
        class="watchlist"
      >
        <img
          alt="watchlist"
          class="action_image"
          src="watchlist.jpg"
        />
        <section
          class="info_section"
        >
          <p
            class="page_info"
          >
            You can keep a track of all the movies or tv shows which you want to watch in future but you need to login first.
          </p>
          <button
            class="visit"
            type="button"
          >
            Visit →
          </button>
        </section>
      </section>
    </section>
    <section
      class="feature_page"
    >
      <section
        class="features"
      >
        <span
          class="features_header"
        >
          Features coming soon
        </span>
        <ul
          class="feature_list"
        >
          <li
            class="feature"
          >
            Subscriptions
          </li>
          <li
            class="feature"
          >
            New Features are yet to add
          </li>
        </ul>
      </section>
      <section
        class="feedback"
      >
        <span
          class="feedback_header"
        >
          Feedback makes A.I.K.A.I perfect
        </span>
        <button
          class="feedback_button"
          type="button"
        >
          Share feedback
        </button>
      </section>
      <span
        class="signature"
      >
        Made by
         
        <a
          class="profile_link"
          href="https://github.com/Salman-Shaik"
          rel="noopener noreferrer"
          target="_blank"
        >
          Salman Shaik
        </a>
      </span>
      <svg
        viewBox="0 0 1440 320"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0,64L48,58.7C96,53,192,43,288,69.3C384,96,480,160,576,160C672,160,768,96,864,106.7C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
          fill="#EB2F53"
          fill-opacity="1"
        />
      </svg>
    </section>
  </section>
</div>
`;
